contracts/origination/OriginationControllerSTIRFRY.sol

```diff
@@ -0,0 +1,268 @@
+     *
+     * @dev The only collateral accepted by the STIRFRY Origination Controller is a Vault Factory vault.
+     *
+     * @param loanTerms                     The terms agreed by the lender and borrower.
+     * @param stirfryData                   The stirfry data to initialize the loan with.
+     * @param vaultAddress                  The address of the vault.
+     * @param borrower                      Address of the borrower.
+     * @param lender                        Address of the lender.
+     *
+     * @return loanId                       The unique ID of the new loan.
+     */
+    function _initialize(
+        LoanLibrary.LoanTerms calldata loanTerms,
+        StirfryData calldata stirfryData,
+        address vaultAddress,
+        address borrower,
+        address lender
+    ) internal nonReentrant returns (uint256 loanId) {
+        // get fee snapshot from fee controller
+        (LoanLibrary.FeeSnapshot memory feeSnapshot) = feeController.getFeeSnapshot();
+
+        // transfer fixed interest amount from borrower to lender's vault
+        IERC20(stirfryData.vaultedCurrency).safeTransferFrom(
+            borrower,
+            vaultAddress,
+            stirfryData.borrowerVaultedCurrencyAmount
+        );
+
+        // collect vault from lender and send to LoanCore
+        IERC721(address(vaultFactory)).transferFrom(lender, address(loanCore), loanTerms.collateralId);
+
+        // create loan in LoanCore
+        loanId = loanCore.startLoan(lender, borrower, loanTerms, feeSnapshot);
+    }
+
+    // ============================================ ADMIN ===============================================
+
+    /**
+     * @notice Whitelist a currency pair to be used for stirfry loans. The first currency is the currency that
+     *         is set in the loan terms. The second currency is the collateral currency.
+     *
+     * @dev The pair ratio is defined as the vaulted currency decimals divided by the collateral currency decimals.
+     *      This ratio is used to compare currency amounts when they use different decimal places. It is very important
+     *      to understand that this ratio is designed such that the decimal places of the loan terms currency must
+     *      be less than the decimal places of the collateral currency. For scenarios where the loan terms currency
+     *      decimals are greater than the collateral currency, input validations will fail.
+     *
+     * @param currency1                  The first currency in the pair.
+     * @param currency2                  The second currency in the pair.
+     * @param ratio                      The ratio of the pair.
+     * @param isAllowed                  Whether the pair is allowed.
+     */
+    function setPair(address currency1, address currency2, uint256 ratio, bool isAllowed) external onlyOwner {
+        bytes32 key = keccak256(abi.encodePacked(currency1, currency2, ratio));
```

Comment:
Although the likelihood of hash collisions is very low in this context, we should be aware that generating keys through hashing could theoretically lead to such issues. 
Given that we are only dealing with a limited number of whitelisted pairs, the risk is minimal here.

Reply:
Yeah, noted on the hash collisions, I would say if there were hundreds of things being whitelisted I would be a little more concerned. But due to the low number of combinations I think will be supported, I had the same reasoning as you that the likelihood is extremely low.